### PR TITLE
Fix graph sequence (atomistic graphs in `initialise_graph_with_metadata` had duplicated residues) 

### LINKED ISSUE #267: graph feature `sequence_{chain_id}` contains duplicate residues for atomistic graphs
I'm trying to extract sequences from graphs loaded from PDB files.   

When constructing a residue-granularity graph from the protein, the sequence (stored at `g.graph[f'sequence_{chain_id}'` is as expected.  However, when using `atom` granularity, the graph's sequence attribute contains repeated residue letters (I am guessing for each atom in the amino acid).  

```python
# Construct graphs, one atomistic and one residue level
g_atom = construct_graph(
    pdb_code="6HD6",
    config=ProteinGraphConfig(
        pdb_dir=pdb_dir,
        granularity="atom", # atomistic 
    )
)

g_res = construct_graph(
    pdb_code="6HD6",
    config=ProteinGraphConfig(
        pdb_dir=pdb_dir,
        granularity="CA", # residue
    )
)

```

```python
c = 'A'
g_res.graph[f'sequence_{c}']
```
```
'AMDPSSPNYDKWEMERTDITMKHKLGGGQYGEVYEG ....
```

```python
g_atom.graph[f'sequence_{c}']
```
```
'AAAAAMMMMMMMMDDDDDDDDPPPPPPPSSSSSSSSSSS ....
```

### PR DESCRIPTION
**Pull request by kamurani**
Fixes #267 

This ensures that atomistic level graphs have the same sequence (for each `c` in `chain_ids`) as those created with `CA` or residue granularity. 

Tested with an assertion (comparing sequences from residue and atomistic graphs).

### CODE REVIEW

**Comment by kamurani** [comment]
@a-r-j  where abouts should I put the tests that I came up with?

**Comment by a-r-j** [comment]
> @a-r-j where abouts should I put the tests that I came up with?

https://github.com/a-r-j/graphein/blob/master/tests/protein/test_graphs.py

File: graphein/protein/graphs.py [hunk 1]
```diff
@@ -493,6 +493,13 @@ def initialise_graph_with_metadata(
             sequence = protein_df.loc[protein_df["chain_id"] == c][
                 "residue_name"
             ].str.cat()
+        elif granularity == "atom":
+            protein_df = protein_df.loc[protein_df["atom_name"] == "CA"]
```
Comment (a-r-j) [inline graphein/protein/graphs.py:497]: Hmm, is keeping the CA subset of `protein_df` not problematic downstream?

Do we not want to store this in a temporary variable (if at all, can we not just add the atom_name selection into L499 when we select by chain ID?

Comment (kamurani) [reply graphein/protein/graphs.py]: Yep, 100%.  My bad, I forgot that when it's passed to the graph construction it only passes a reference to it.

**Pull request merged**